packages/js/src/analysis/plugins/shortcode-plugin.js

```diff
@@ -0,0 +1,324 @@
+/* global tinyMCE */
+/* global wpseoScriptData */
+/* global ajaxurl */
+/* global _ */
+
+const shortcodeNameMatcher = "[^<>&/\\[\\]\x00-\x20=]+?";
+const shortcodeAttributesMatcher = "( [^\\]]+?)?";
+
+const shortcodeStartRegex = new RegExp( "\\[" + shortcodeNameMatcher + shortcodeAttributesMatcher + "\\]", "g" );
+const shortcodeEndRegex = new RegExp( "\\[/" + shortcodeNameMatcher + "\\]", "g" );
```

Comment:
Let's use capitalized variable names for constants, e.g. `SHORTCODE_START_REGEX`. See https://javascript.info/variables#uppercase-constants